process/block/preprocess/transactions.go

```diff
@@ -630,14 +629,37 @@ func (txs *transactions) processTxsFromMe(
 		txsFromMe,
 	)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 
 	if !haveTime() {
-		return process.ErrTimeIsOut
+		return nil, nil, process.ErrTimeIsOut
+	}
+
+	return calculatedMiniBlocks, mapSCTxs, nil
+}
+
+func (txs *transactions) processTxsFromMeAndCreateScheduled(
+	body *block.Body,
+	haveTime func() bool,
+	randomness []byte,
+) error {
+	calculatedMiniBlocks, mapSCTxs, err := txs.processTxsFromMe(body, haveTime, randomness)
+	if err != nil {
+		return err
+	}
+
+	isShardStuckFalse := func(uint32) bool {
```

Comment:
these 3 functions could have been declared as vars as we might reuse them in other functions

Reply:
done